draft-kazuho-httpbis-priority.md

```diff
@@ -112,33 +112,93 @@ Unknown parameters MUST be ignored.
 
-* `blocking` indicates that the response prevents other responses from being
-  used.
-* `document` indicates that the response contains the document that is being
-  processed.
-* `non-blocking` indicates that the response does not prevent the client from
-  using the document even though the response is being used or referred to by
-  the document.
+| Urgency         | Definition                      |
+|----------------:|:--------------------------------|
+|              -1 | blocking ({{blocking}})         |
+|               0 | document ({{document}})         |
+| between 1 and 5 | non-blocking ({{non-blocking}}) |
+|               6 | background ({{background}})     |
+{: #urgencies title="Urgencies"}
 
-The default value is `document`.
+The value is encoded as an sh-integer.  The default value is zero.
 
-A server SHOULD transmit HTTP responses in the order of their urgency:
-`blocking` first, followed by `document`, followed by `non-blocking`.
+A server SHOULD transmit HTTP responses in the order of their urgency values.
+The lesser the value is, the higher the precedence is.
 
 The following example shows a request for a CSS file with the urgency set to
-`blocking`:
+`-1`:
 
 ~~~ example
 :method = GET
 :scheme = https
 :authority = example.net
 :path = /style.css
-priority = urgency=blocking
+priority = urgency=-1
 ~~~
 
+The definition of the urgencies and their expected use-case are described below.
+Endpoints SHOULD respect the definition of the values when assigning urgencies.
+
+### blocking
+
+The blocking urgency indicates that the response prevents other responses from
+being used.
+
+Value of -1 is used to represent this urgency.
+
+For example, a stylesheet being referred from the HEAD section of an HTML
+document blocks a web browser from rendering the HTML.  In such case, the
+stylesheet is given the blocking urgency.
+
+### document
+
+The document urgency indicates that the response contains the document that is
+being processed.  This urgency is also used for responses that deserve the same
+precedence as the contents of the document.
+
+Value of zero (0) is used to represent this urgency.
+
+For example, when a user using a web browser navigates to a different HTML
+document, the request for that HTML is given the document urgency.  When that
+HTML document uses a custom font, the request for that custom font SHOULD also
+be given the document urgency.  This is because the availablity of the custom
+font is the precondition for the user to use the portion of the HTML document,
+which is to be rendered by that font.
+
+### non-blocking
+
+The non-blocking urgency indicates that the response does not prevent the client
+from using the document even though the response is being incorporated to or
+referred to by the document.
+
+For example, inline images typically improve the usability of an HTML document.
+Therefore, such images will be given the non-blocking urgency.
```

Comment:
As remarked by Lucas before, this passage feels weird... what exactly are "inline" images here? And how does them improving the usability of a document lead to them being non-blocking?

Suggestion:
For example, images referred to in an HTML document can typically be processed and rendered independently from other resources. Therefore, such images will be given the non-blocking urgency.

Reply:
I'm fine with explaining what "inline images" mean here.

OTOH, I think we need to explain why a custom font, which typically blocks the rendering of some text, should be given higher precedence than an inline image.

It has been my assumption that some images being unavailable is considered to have less impact than some text being unavailable, and that is because an HTML document is considered to be somewhat usable even though images are yet to be rendered.

I could well be wrong, but we need to discuss the rationale, if we want to suggest that fonts should be given higher precedence than images.

Reply:
I would say your assumptions are correct and that fonts need higher precedence than images.
I just didn't feel the text made that very clear with the "improve the usability" part. 

Besides that, I also feel that people reading this will probably understand that fonts can block text rendering and that text is typically more important than images without explicitly stating it.

Reply:
Thank you for your thoughts. I've expanded why images are improvements. PTAL at 68f9660.